src/charm.py

```diff
@@ -1406,6 +1436,7 @@ def _reconciliation(self, _event: EventBase) -> None:
             self.unit.status = WaitingStatus("Waiting for storage")
             _event.defer()
             return
+        self._change_uploads_directory_ownership()
```

Comment:
I'm wondering if changing uploads directory ownership needs to run every reconcile, can be discussed for later issues though!

Reply:
Agree, maybe this could be discussed in the story ISD-721 (refactor). I added a link to this comment there, thanks!